app/controllers/pet_applicants_controller.rb

```diff
@@ -0,0 +1,9 @@
+class PetApplicantsController < ApplicationController
+  def index
+  end
+
+  def create
+    PetApplicant.create!(pet: Pet.find(params[:pet]), applicant: Applicant.find(params[:id]))
```

Comment:
Boom! You already have this controller made! So in order to get your application to arrive at this action, you would need to update your button on the applicant show page to be:
`<%= button_to "Adopt this Pet", "/applicants/#{@applicant.id}/pets/#{result.id}", method: :post %>`
Note that you don't need to add any Turbo business because Turbo actually doesn't affect buttons! Just links and forms. 

For this to work, you'd also need to update your routes. Seem my comment in your routes file